src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/AppealSubmissionTemplate.java

```diff
@@ -64,6 +64,76 @@ public Map<String, Object> mapFieldValues(
         fieldValues.put("appellantDateOfBirth", formatDateForRendering(asylumCase.read(APPELLANT_DATE_OF_BIRTH, String.class).orElse("")));
         fieldValues.put("appellantTitle", asylumCase.read(APPELLANT_TITLE, String.class).orElse(""));
 
+        YesOrNo isDetained = asylumCase.read(APPELLANT_IN_DETENTION, YesOrNo.class).orElse(null);
+        fieldValues.put("appellantInDetention", isDetained);
+        if (isDetained.equals(YesOrNo.YES)) {
+            StringBuilder sb = new StringBuilder("Detained");
+            YesOrNo isAcceleratedDetainedAppeal = asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class).orElse(null);
+            if (isAcceleratedDetainedAppeal.equals(YesOrNo.YES)) {
+                sb.append(" - Accelerated");
+                fieldValues.put("isAcceleratedDetainedAppeal", YesOrNo.YES);
+            }
+            fieldValues.put("detentionStatus", sb.toString());
+            sb.setLength(0);
+
+            String detentionFacility = asylumCase.read(DETENTION_FACILITY, String.class).orElse(null);
+            String detentionFacilityName = "";
+            switch (detentionFacility) {
```

Comment:
Just a suggestion, can you take the block related to setting fields of detention facility and move to a private method?